collector/test/CollectorConfigTest.cpp

```diff
@@ -145,4 +151,71 @@ TEST(CollectorConfigTest, TestEnableExternalIpsRuntimeConfig) {
   EXPECT_TRUE(config.EnableExternalIPs());
 }
 
+TEST(CollectorConfigTest, TestYamlConfigToConfigMultiple) {
+  std::vector<std::pair<std::string, bool>> tests = {
+      {R"(
+                  networkConnectionConfig:
+                    enableExternalIps: true
+               )",
+       true},
+      {R"(
+                  networkConnectionConfig:
+                    enableExternalIps: false
+               )",
+       false},
+      {R"(
+                  networkConnectionConfig:
+               )",
+       false},
+      {R"(
+                  networkConnectionConfig:
+                    unknownField: asdf
+               )",
+       false}};
+
+  for (const auto& [yamlStr, expected] : tests) {
+    YAML::Node yamlNode = YAML::Load(yamlStr);
+
+    MockCollectorConfig config;
+
+    bool result = config.MockYamlConfigToConfig(yamlNode);
+    auto runtime_config = config.GetRuntimeConfig();
+
+    EXPECT_TRUE(result);
+    EXPECT_TRUE(runtime_config.has_value());
+
+    if (runtime_config.has_value()) {
```

Comment:
You don't need the if, you already expected `runtime_config.has_value()` to be true.
```suggestion
    EXPECT_TRUE(runtime_config.has_value());
```

Reply:
Done